packages/core/tests/parameter/flag/formatting.spec.ts

```diff
@@ -488,6 +539,32 @@ describe("formatDocumentationForFlagParameters", function () {
             compareToBaseline(this, StringArrayBaselineFormat, lines);
         });
 
+        it("required parsed flag with default from env var", function () {
+            // GIVEN
+            type Positional = [];
+            type Flags = {
+                readonly requiredParsed: string;
+            };
+
+            const parameters: TypedCommandParameters<Flags, Positional, CommandContext> = {
+                flags: {
+                    requiredParsed: {
+                        kind: "parsed",
+                        parse: String,
+                        default: { env: "REQUIRED_PARSED" },
+                        brief: "required parsed flag",
+                    },
+                },
+                positional: { kind: "tuple", parameters: [] },
+            };
+
+            // WHEN
+            const lines = formatDocumentationForFlagParameters(parameters.flags, parameters.aliases ?? {}, defaultArgs);
+
+            // THEN
+            compareToBaseline(this, StringArrayBaselineFormat, lines);
+        });
+
```

Comment:
Can we also add a `redact: true` test?

Reply:
Oh at the moment there isn't any difference in the help text when `redact: true` is enabled. Should there be?

```text
    [--requiredBoolean/--noRequiredBoolean]  required boolean flag           [default = <from env REQUIRED_BOOLEAN>]
 -h --help                                   Print help information and exit
```

Reply:
At the moment the help text doesn't display the value of the environment variable, only the name. While I think it would be incredibly handy to print the value (or redacted value) we also still need to print the name and all of that takes up a lot of space. Not sure what the best display would be yet.

Reply:
Which of these options would be the best for displaying the default value. Right now we have the last one, which has the added benefit of not needing to read from the process to build the help text.

```text
    [--flag]  some flag           [default = ACTUAL_VALUE]
    [--flag]  some flag           [default = ACTUAL_VALUE <from env REQUIRED_BOOLEAN>]
    [--flag]  some flag           [default = <from env REQUIRED_BOOLEAN>]
```

With the following for `redact: true`

```text
    [--flag]  some flag           [default = ************]
    [--flag]  some flag           [default = ************ <from env REQUIRED_BOOLEAN>]
    [--flag]  some flag           [default = <from env REQUIRED_BOOLEAN>]
```

Reply:
My preference would be option (2) followed by (1) if length is a concern.

Reply:
Sounds good, we can do 2 and still have the backout to 3 if there is no value set.